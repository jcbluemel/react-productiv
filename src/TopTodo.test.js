import React from "react";
import { render } from "@testing-library/react";
import TopTodo from "./TopTodo";

const TEST_TODOS = [
  {
  id: "1",
  title: "Test1",
  description: "test1",
  priority: 1
  },
  {
    id: "2",
    title: "Test2",
    description: "test2",
    priority: 2
  }
];

it("renders without crashing", function() {
  render(<TopTodo todos={TEST_TODOS} />);
});

it("matches snapshot", function() {
  const { asFragment } = render(<TopTodo todos={TEST_TODOS} />);
  expect(asFragment()).toMatchSnapshot();
});

it("returns todo with highest priority", function() {
  const { container } = render(<TopTodo todos={TEST_TODOS} />);
  expect(container).toContainHTML('class="Todo" id="1"');
  expect(container).toContainHTML('<b>Test1</b>');
  expect(container).toContainHTML('<small>Priority: 1</small>');
  expect(container).toContainHTML('<small>test1</small>');
});