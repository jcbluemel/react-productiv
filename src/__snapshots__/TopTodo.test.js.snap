// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Todo"
    id="1"
  >
    <div>
      <b>
        Test1
      </b>
       
      <small>
        Priority: 1
      </small>
    </div>
    <div>
      <small>
        test1
      </small>
    </div>
  </div>
</DocumentFragment>
`;
